plugin/core/sessions.py

```diff
@@ -220,24 +275,49 @@ def clear_dotted_value(current: dict, dotted: str) -> None:
     current.pop(keys[-1], None)
 
 
-class Session(object):
-    def __init__(self,
-                 config: ClientConfig,
-                 workspace_folders: List[WorkspaceFolder],
-                 client: Client,
-                 on_pre_initialize: 'Optional[Callable[[Session], None]]' = None,
-                 on_post_initialize: 'Optional[Callable[[Session], None]]' = None,
-                 on_post_exit: Optional[Callable[[str], None]] = None) -> None:
+class Session(Client):
+
+    @classmethod
+    def name(cls) -> str:
+        raise NotImplementedError()
+
+    @classmethod
+    def needs_update_or_installation(cls) -> bool:
+        return False
+
+    @classmethod
+    def install_or_update(cls) -> None:
+        pass
+
+    @classmethod
+    def standard_configuration(cls) -> ClientConfig:
```

Comment:
I'd like to be able to softly transition to this new API by switching out `NpmClientHandler` without changing subclasses of it. The fact that this is a class method rather than an instance method might make it hard, messy, or even impossible.

Previously `config` was an instance property. What do you think about trying to make this an instance class?

Reply:
Hmm, we'll need config to create session...

Reply:
I was able to implement everything. Even if it was a bit hacky due to needing to initialize server resources from `standard_configuration`.

Reply:
I was also playing around with lsp_utils, see my draft pull request there. I was thinking to re-purpose the name LanguageHandler to be a temporary API wrapper class.